packages/jsActions/nanoflow-actions-native/src/permissions/RequestGenericPermission.ts

```diff
@@ -0,0 +1,65 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import { Alert, Platform } from "react-native";
+import { check, request, RESULTS, openSettings, PERMISSIONS, Permission } from "react-native-permissions";
+import { ANDROIDPermissionName, IOSPermissionName } from "../../typings/RequestGenericPermission";
+
+// BEGIN EXTRA CODE
+
+function handleBlockedPermission(): void {
+    Alert.alert("title", "need to go to settings for activation", [
+        // draft text
+        { text: "go to settings", onPress: () => openSettings() },
+        { text: "cancel" }
+    ]);
+}
+
+function mapPermissionName(permissionName: string): Permission | undefined {
+    if (Platform.OS === "ios") {
+        const nameWithoutSuffix = permissionName.replace("_IOS", "") as IOSPermissionName;
+
+        return PERMISSIONS.IOS[nameWithoutSuffix];
+    } else if (Platform.OS === "android") {
+        const nameWithoutSuffix = permissionName.replace("_ANDROID", "") as ANDROIDPermissionName;
+
+        return PERMISSIONS.ANDROID[nameWithoutSuffix];
+    }
+}
+// END EXTRA CODE
+
+/**
+ * @param {"NanoflowCommons.Enum_Permissions.CAMERA"|"NanoflowCommons.Enum_Permissions.CONTACTS"} permission - This field is required.
+ * @returns {Promise.<"NanoflowCommons.Enum_PermissionStatus.unavailable"|"NanoflowCommons.Enum_PermissionStatus.denied"|"NanoflowCommons.Enum_PermissionStatus.limited"|"NanoflowCommons.Enum_PermissionStatus.granted"|"NanoflowCommons.Enum_PermissionStatus.blocked">}
+ */
+export async function GenericPermission(
+    permission: string
+): Promise<"unavailable" | "blocked" | "denied" | "granted" | "limited"> {
+    // BEGIN USER CODE
+    if (!permission) {
+        return Promise.reject(new Error("Input parameter 'permission' is required"));
+    }
+    const mappedPermissionName = mapPermissionName(permission);
+    if (!mappedPermissionName) {
+        console.error(`${permission} permission is not found`);
+        return Promise.resolve("unavailable");
+    }
+    const permissionStatus = await check(mappedPermissionName!);
+
+    switch (permissionStatus) {
+        case RESULTS.GRANTED:
+        case RESULTS.UNAVAILABLE:
+        case RESULTS.LIMITED:
+            return permissionStatus;
+        case RESULTS.BLOCKED:
+            handleBlockedPermission();
+            return RESULTS.BLOCKED;
+        case RESULTS.DENIED:
+            return request(mappedPermissionName!);
```

Comment:
What happens if this request is granted/accepted? How is the result returned/handled by the calling Nanoflow?

I.e. if this expression is reached, and the request is blocked, the `handleBlockedPermission` wouldn't be called.

Reply:
if the permission status is denied and then we are able to request it from the user. `request` function returns the result from this request, it can be denied, granted or blocked. In all cases the JsAction should return the result of this request. we shouldn't call `handleBlockedPermission` if the result of the request is blocked. we do that only if the permission status is blocked due to the user has denied it before in a previous request.